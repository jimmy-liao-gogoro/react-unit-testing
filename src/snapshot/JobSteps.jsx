import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import './styles.css';

const JobSteps = ({ step, total }) => {
  if (step === null) {
    return null;
  }

  const stepItems = [];
  for (let i = 1; i <= total; i += 1) {
    const circleClass = classnames('circle', {
      'circle-active': step === i,
    });
    stepItems.push(
      <span className="step" key={i}>
        {i !== total ? <span className="line" /> : null}
        <span className={circleClass}>
          {i}
        </span>
      </span>,
    );
  }

  return <span className="container">{stepItems}</span>;
};

JobSteps.propTypes = {
  step: PropTypes.number,
  total: PropTypes.number.isRequired,
};

JobSteps.defaultProps = {
  step: null,
};

export default JobSteps;
