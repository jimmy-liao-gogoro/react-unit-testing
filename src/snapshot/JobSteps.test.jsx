import React from 'react';
import { create } from 'react-test-renderer';

import JobSteps from './JobSteps';

describe('<JobSteps/>', () => {
  it('should render nothing if step is null', () => {
    const tree = create(<JobSteps step={null} total={3} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should highlight step 1 if job is in step 1', () => {
    const tree = create(<JobSteps step={1} total={3} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
