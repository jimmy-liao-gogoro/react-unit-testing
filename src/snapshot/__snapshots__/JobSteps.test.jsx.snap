// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobSteps/> should highlight step 1 if job is in step 1 1`] = `
<span
  className="container"
>
  <span
    className="step"
  >
    <span
      className="line"
    />
    <span
      className="circle circle-active"
    >
      1
    </span>
  </span>
  <span
    className="step"
  >
    <span
      className="line"
    />
    <span
      className="circle"
    >
      2
    </span>
  </span>
  <span
    className="step"
  >
    <span
      className="circle"
    >
      3
    </span>
  </span>
</span>
`;

exports[`<JobSteps/> should render nothing if step is null 1`] = `null`;
